cmd/xl-v1-utils.go

```diff
@@ -382,35 +382,37 @@ func evalDisks(disks []StorageAPI, errs []error) []StorageAPI {
 	return newDisks
 }
 
-// Errors specifically generated by getPartSizeFromIdx function.
+// Errors specifically generated by calculatePartSizeFromIdx function.
 var (
 	errPartSizeZero  = errors.New("Part size cannot be zero")
 	errPartSizeIndex = errors.New("Part index cannot be smaller than 1")
 )
 
-// getPartSizeFromIdx predicts the part size according to its index. It also
-// returns -1 when totalSize is -1.
-func getPartSizeFromIdx(totalSize int64, partSize int64, partIndex int) (int64, error) {
+// calculatePartSizeFromIdx calculates the part size according to input index.
+// returns error if totalSize is -1, partSize is 0, partIndex is 0.
+func calculatePartSizeFromIdx(totalSize int64, partSize int64, partIndex int) (int64, error) {
+	if totalSize < 0 {
+		return 0, traceError(errInvalidArgument)
+	}
 	if partSize == 0 {
 		return 0, traceError(errPartSizeZero)
 	}
 	if partIndex < 1 {
 		return 0, traceError(errPartSizeIndex)
 	}
-	switch totalSize {
-	case -1, 0:
-		return totalSize, nil
-	}
-	// Compute the total count of parts
-	partsCount := totalSize/partSize + 1
-	// Return the part's size
-	switch {
-	case int64(partIndex) < partsCount:
-		return partSize, nil
-	case int64(partIndex) == partsCount:
-		// Size of last part
-		return totalSize % partSize, nil
-	default:
-		return 0, nil
+	if totalSize > 0 {
+		// Compute the total count of parts
+		partsCount := totalSize/partSize + 1
+		// Return the part's size
+		switch {
+		case int64(partIndex) < partsCount:
+			totalSize = partSize
```

Comment:
Can you add a new variable `currPartSize` and avoid setting totalSize because its meaning is changed here

Reply:
Will do